### ISSUE TITLE
middle() on affine units

### ISSUE THREAD

**Issue by jariji** [PainterQubits/Unitful.jl#629]
Should I be able to do this?

```jl
julia> middle(33°F, 35°F)
ERROR: AffineError: an invalid operation was attempted with affine units: °F
```

**Comment by cstjean** [comment]
Yeah, several similar methods are probably necessary https://github.com/PainterQubits/Unitful.jl/issues/200

**Comment by jariji** [comment]
I was thinking the solution was to just redefine `Statistics.middle(x,y) = (x-y)/2 + y` but 

```jl
julia> (33°F - 32°F)/2  + 32°F
273.4277777777778 K
```

produces a result in Kelvin instead of Fahrenheit, which I didn't expect.

**Comment by cstjean** [comment]
Yeah, I think a lot of people aren't super happy with quickly temperature manipulation becomes all-Kelvin. You can convert it back to Celsius, of course.

**Comment by jariji** [comment]
Related https://github.com/JuliaStats/Statistics.jl/issues/47
